### ISSUE TITLE
Enigma plugin gives HTTP 500 error when clicking PGP Keys

### ISSUE THREAD

**Issue by ssurfertuga** [roundcube/roundcubemail#7024]
So I'm using RC for some time now and I decided to start using enigma to have my emails signed and encrypted as necessary.
I followed all the standard steps to install the plugin (just like any other plugin) but, as soon as I click "PGP Keys" on settings, I get an HTTP 500 error on Chrome.

This is what I could find in the logs:

Nov 12 14:38:18 mail roundcube: PHP Fatal error:  Uncaught Error: Call to undefined method rcube_config::keyservers() in /opt/www/roundcubemail-1.3.8/plugins/enigma/lib/enigma_ui.php:141#012Stack trace:#012#0 /opt/www/roundcubemail-1.3.8/plugins/enigma/lib/enigma_ui.php(43): enigma_ui->add_js()#012#1 /opt/www/roundcubemail-1.3.8/plugins/enigma/enigma.php(423): enigma_ui->init()#012#2 /opt/www/roundcubemail-1.3.8/program/lib/Roundcube/rcube_plugin_api.php(493): enigma->preferences_ui()#012#3 /opt/www/roundcubemail-1.3.8/index.php(289): rcube_plugin_api->exec_action('plugin.enigmake...')#012#4 {main}#012  thrown in /opt/www/roundcubemail-1.3.8/plugins/enigma/lib/enigma_ui.php on line 141
Nov 12 14:38:27 mail roundcube: PHP Fatal error:  Uncaught Error: Call to undefined method rcube_config::keyservers() in /opt/www/roundcubemail-1.3.8/plugins/enigma/lib/enigma_ui.php:141#012Stack trace:#012#0 /opt/www/roundcubemail-1.3.8/plugins/enigma/lib/enigma_ui.php(43): enigma_ui->add_js()#012#1 /opt/www/roundcubemail-1.3.8/plugins/enigma/enigma.php(423): enigma_ui->init()#012#2 /opt/www/roundcubemail-1.3.8/program/lib/Roundcube/rcube_plugin_api.php(493): enigma->preferences_ui()#012#3 /opt/www/roundcubemail-1.3.8/index.php(289): rcube_plugin_api->exec_action('plugin.enigmake...')#012#4 {main}#012  thrown in /opt/www/roundcubemail-1.3.8/plugins/enigma/lib/enigma_ui.php on line 141
Nov 12 14:38:28 mail roundcube: PHP Fatal error:  Uncaught Error: Call to undefined method rcube_config::keyservers() in /opt/www/roundcubemail-1.3.8/plugins/enigma/lib/enigma_ui.php:141#012Stack trace:#012#0 /opt/www/roundcubemail-1.3.8/plugins/enigma/lib/enigma_ui.php(43): enigma_ui->add_js()#012#1 /opt/www/roundcubemail-1.3.8/plugins/enigma/enigma.php(423): enigma_ui->init()#012#2 /opt/www/roundcubemail-1.3.8/program/lib/Roundcube/rcube_plugin_api.php(493): enigma->preferences_ui()#012#3 /opt/www/roundcubemail-1.3.8/index.php(289): rcube_plugin_api->exec_action('plugin.enigmake...')#012#4 {main}#012  thrown in /opt/www/roundcubemail-1.3.8/plugins/enigma/lib/enigma_ui.php on line 141
Nov 12 14:38:29 mail roundcube: PHP Fatal error:  Uncaught Error: Call to undefined method rcube_config::keyservers() in /opt/www/roundcubemail-1.3.8/plugins/enigma/lib/enigma_ui.php:141#012Stack trace:#012#0 /opt/www/roundcubemail-1.3.8/plugins/enigma/lib/enigma_ui.php(43): enigma_ui->add_js()#012#1 /opt/www/roundcubemail-1.3.8/plugins/enigma/enigma.php(423): enigma_ui->init()#012#2 /opt/www/roundcubemail-1.3.8/program/lib/Roundcube/rcube_plugin_api.php(493): enigma->preferences_ui()#012#3 /opt/www/roundcubemail-1.3.8/index.php(289): rcube_plugin_api->exec_action('plugin.enigmake...')#012#4 {main}#012  thrown in /opt/www/roundcubemail-1.3.8/plugins/enigma/lib/enigma_ui.php on line 141

I already made sure the directory for the keys has the right permissions. Maybe some dependency is missing?
Please I need help here! 
Thanks!

**Comment by ssurfertuga** [comment]
Trying to solve this error I also found useful to post that the URL on the request is (xxx for redacted):
"https://mail.xxx.eu/mail/?_task=settings&_action=plugin.enigmakeys"

Response: HTTP ERROR 500

In other words, I can't even see the page column for creating keys.

**Comment by smares** [comment]
The keyservers() method seems to have been introduced with Roundcube 1.4.0 and your paths indicate that you are using Roundcube 1.3.8. Did you only update the plugin without updating the Roundcube base?

**Comment by ssurfertuga** [comment]
> Did you only update the plugin without updating the Roundcube base?

Actually I have RC 1.3.8 installed and I just added the plugin yesterday. It wasn't upgraded, just copied into plugins folder.
However I got the plugin from GitHub repository for RC 1.4.0.
Is that a problem?

**Comment by smares** [comment]
Yes, that's a problem as you can see. :) Either upgrade Roundcube to 1.4.0 or use the plugin from the 1.3.8 tag: https://github.com/roundcube/roundcubemail/tree/1.3.8

**Comment by ssurfertuga** [comment]
> Yes, that's a problem as you can see. :) Either upgrade Roundcube to 1.4.0 or use the plugin from the 1.3.8 tag: https://github.com/roundcube/roundcubemail/tree/1.3.8

It never crossed my mind that the plugin version wasn't independent from the RC version... Thanks for the headsup. Will try now and report back!
Thanks

**Comment by ssurfertuga** [comment]
> Either upgrade Roundcube to 1.4.0 or use the plugin from the 1.3.8 tag

It worked! Now I can click on "PGP Keys" in Settings.
However when I try to create a key pair I get the message "Your web browser does not support cryptography. Unable to generate a key pair!"

This happens on the latest versions of Safari and Chrome for MacOS.
Any ideas?

**Comment by alecpl** [comment]
Check browser console for errors. If you open plugins/enigma/openpgp.min.js file what version do you see?

**Comment by ssurfertuga** [comment]
Checked the browser console and this error pops up:
"GET https://mail.hugoramos.eu/mail/plugins/enigma/openpgp.min.js net::ERR_ABORTED 404 (Not Found)"

After much research I found out that my install was missing some files because it was installed by iRedMail stack. All is fine now. I just don't understand why iRedMail uses a short version (not all the files) of Roundcube... 

Thanks for all the help!
